python/ray/train/lightning/lightning_trainer.py

```diff
@@ -0,0 +1,309 @@
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks import ModelCheckpoint
+import logging
+
+from typing import TYPE_CHECKING, Callable, Dict, Optional, Union, Type, Any
+from inspect import isclass
+from pytorch_lightning.plugins.environments import ClusterEnvironment
+
+import ray
+from ray.air import session
+from ray.air.checkpoint import Checkpoint
+from ray.air.config import CheckpointConfig, DatasetConfig, RunConfig, ScalingConfig
+from ray.data.preprocessor import Preprocessor
+from ray.train.data_parallel_trainer import DataParallelTrainer
+from ray.train.torch.config import TorchConfig
+from ray.train.trainer import GenDataset
+from ray.train.constants import (
+    EVALUATION_DATASET_KEY,
+    TRAIN_DATASET_KEY,
+)
+from torch.utils.data import DataLoader
+from ray.train.lightning.lightning_utils import (
+    RayDDPStrategy, RayEnvironment, RayModelCheckpoint, RayDataModule
+)
+from ray.util import PublicAPI
+import ray.train as train
+
+logger = logging.getLogger(__name__)
+
+LIGHTNING_MODULE_KEY = "_lightning_module"
+LIGHTNING_MODULE_CONFIG_KEY = "_lightning_module_config"
+LIGHTNING_TRAINER_CONFIG_KEY = "_lightning_trainer_config"
+LIGHTNING_TRAINER_FIT_PARAMS_KEY = "_lightning_trainer_fit_params"
+MODEL_CHECKPOINT_CONFIG_KEY = "_model_checkpoint_config"
+DDP_STRATEGY_CONFIG_KEY = "_ddp_strategy_config"
+RAY_DATASET_ITER_CONFIG_KEY = "_ray_dataset_iter_config"
+
+
+@PublicAPI(stability="alpha")
+class LightningTrainer(DataParallelTrainer):
```

Comment:
this should be subclassing TorchTrainer right?

Reply:
Yes. Will change.